src/steps/filter.py

```diff
@@ -112,6 +118,15 @@ def wait_for_subscriptions_on_main_nodes(self, content_topic_list, pubsub_topic=
         assert filter_sub_response["requestId"] == request_id
         assert filter_sub_response["statusDesc"] in ["OK", ""]  # until https://github.com/waku-org/nwaku/issues/2286 is fixed
 
+    @retry(stop=stop_after_delay(70), wait=wait_fixed(1), reraise=True)
+    @allure.step
+    def subscribe_optional_filter_nodes(self, content_topic_list, pubsub_topic=None):
```

Comment:
What's an optional filter node?

Reply:
Most of the tests written so far are using 2 nodes (main_nodes). Ex 1 nwaku and 1 gowaku. This way the test is easier to write and maintain. 
But the framework should support tests with dynamic number of nodes (see test_multiple_nodes) where a test can run with as many nodes as we want. That's why I introduced such methods that can be used in tests that work with more than 2 nodes.